File: Java/TermForge/src/main/java/com/Miguel_Lamazares/TermForge/Validator/Numbers.java
package com.Miguel_Lamazares.TermForge.Validator;

import com.Miguel_Lamazares.TermForge.Animation.*;
import com.Miguel_Lamazares.TermForge.Core.*;


import java.util.Scanner;

public class Numbers {

    public static int onlyNumbers(java.util.Scanner scanner) {
        int choice;
        while (true) {
            if (!scanner.hasNextInt()) {
                scanner.nextLine();
                System.out.print("\033[1A");
                System.out.print("\033[2K");

                continue;
            }
            choice = scanner.nextInt();
            scanner.nextLine();

            return choice;
        }
    }

    public static int numbersBetween(java.util.Scanner scanner, int minvalue, int maxvalue) {
        int choice;
        while (true) {
            if (!scanner.hasNextInt()) {
                scanner.nextLine();
                System.out.print("\033[1A");
                System.out.print("\033[2K");

                continue;
            }
            choice = scanner.nextInt();
            scanner.nextLine();

            if (choice <= minvalue - 1) {
                System.out.print("\033[1A");
                System.out.print("\033[2K");
                continue;
            }
            if (choice >= maxvalue) {
                System.out.print("\033[1A");
                System.out.print("\033[2K");
                continue;
            }

            return choice;
        }

    }

}

File: Python /termforge/validator/numbers.py
import sys

def onlyNumbers() -> int:
    while True:
        user_input = input()

        if not user_input.strip().isdigit():
            sys.stdout.write("\033[F")
            sys.stdout.write("\033[K")
            continue

        choice = int(user_input)

        return choice
        
def numbersBetween(Min_value: int, max_value: int) -> int:
    while True:
        user_input = input()

        if not user_input.strip().isdigit():
            sys.stdout.write("\033[F")
            sys.stdout.write("\033[K")
            continue

        choice = int(user_input)

        if choice <= Min_value - 1:
            sys.stdout.write("\033[F")
            sys.stdout.write("\033[K")
            continue

        if choice >= max_value + 1:
            sys.stdout.write("\033[F")
            sys.stdout.write("\033[K")
            continue

        return choice